src/transformers/file_utils.py

```diff
@@ -297,6 +297,20 @@ def is_pandas_available():
     return importlib.util.find_spec("pandas") is not None
 
 
+def is_sagemaker_distributed_available():
```

Comment:
I'd rename it to `is_sagemaker_distributed_enabled()` since this is what it does.

it can be available but not enabled, which aren't the same.

Reply:
apologies, I'm late to the party

Reply:
No, this is inconsistent with other: `is_tpu_available()` for instance.

Reply:
Why? I presume you meant to say `is_torch_tpu_available`:
```
def is_torch_tpu_available():
    if not _torch_available:
        return False
    # This test is probably enough, but just in case, we unpack a bit.
    if importlib.util.find_spec("torch_xla") is None:
        return False
    if importlib.util.find_spec("torch_xla.core") is None:
        return False
    return importlib.util.find_spec("torch_xla.core.xla_model") is not None
```
here there is no user input - it just checks if the system has the right bits, and says available/or not.

In `is_sagemaker_distributed_available` there is user input - i.e. the user should be able to turn this functionality off. So sagemaker could be available but not enabled, does it make sense?

Reply:
Or perhaps there isn't a user input here, I'm not familiar with this env, and then you're correct.

but there should be a way for a user to turn it off, are we on the same page for this last bit?

Reply:
In which case your name is then perfectly correct, but how does a user go then about not enabling this feature if they don't want to - I guess that's what I'm missing.

Reply:
No there is no user input for `is_sagemaker_distributed_available` -> it's detecting if the script was launched by sagemaker with the distributed option. The user enables/disables this feature outside of Transformers in the notebook they use to launch that script (it's an option in the sagemaker estimator). Kind of the same way they decide to launch their script on TPUs, which is why I was picking `is_torch_tpu_available()` as a comparison.

Reply:
Thank you for explaining this to me. I incorrectly assumed that the env var the code read was signalling to the framework whether to enable this dist or not.

In general to me:
- `available`: installed/configured.
- `enabled`:  someone actually activated it and wants to use it.

so perhaps these mnemonics are somewhat different in first place to what they mean.